### PR TITLE
Attempt to fix inability to logout from some instances (subdomains)

### PR DESCRIPTION
**Pull request by alectrocute**
## Description

- Attempt to fix inability to logout from some instances (subdomains)
- Turn auth cookie setting/clearing into module actions for more visiblity
- Extract cookie name into config export

## Screenshots

N/A

### CODE REVIEW

File: src/shared/utils/browser/clear-auth-cookie.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import * as cookie from "cookie";
+import { authCookieName } from "shared/config";
+
+export default function clearAuthCookie() {
+  document.cookie = cookie.serialize(authCookieName, "", {
+    expires: new Date("Thu, 01 Jan 1970 00:00:01 GMT"),
```
Comment (alectrocute) [inline src/shared/utils/browser/clear-auth-cookie.ts:6]: Adding `expires` attribute with value way in the past, which seems to be a common pattern in other code bases.

File: src/shared/utils/browser/clear-auth-cookie.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    maxAge: -1,
```
Comment (alectrocute) [inline src/shared/utils/browser/clear-auth-cookie.ts:7]: Changed from `0` to `-1`.

Comment (SleeplessOne1917) [reply src/shared/utils/browser/clear-auth-cookie.ts:7]: What is the point of having both an `expires` and a `maxAge`?

File: src/shared/utils/browser/set-auth-cookie.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import { authCookieName } from "shared/config";
+
+export default function setAuthCookie(jwt: string) {
+  const expires = new Date();
+  expires.setDate(expires.getDate() + 365);
+
+  document.cookie = cookie.serialize(authCookieName, jwt, {
+    expires,
+    secure: isHttps(),
+    domain: location.hostname,
```
Comment (SleeplessOne1917) [inline src/shared/utils/browser/set-auth-cookie.ts:12]: Should probably get rid of domain. Even if it's not causing the issue, I don't think we need it.

File: src/shared/utils/browser/clear-auth-cookie.ts (hunk 2 above)
Comment (SleeplessOne1917) [reply src/shared/utils/browser/clear-auth-cookie.ts:7]: I found [something relevant](https://mrcoles.com/blog/cookies-max-age-vs-expires/):

> **Q. What if I set both expires and max-age in a cookie?**
A. Every browser that supports max-age will ignore the expires regardless of it’s value, and likewise, Internet Explorer will ignore the max-age and just use expires.

Comment (alectrocute) [reply src/shared/utils/browser/clear-auth-cookie.ts:6]: Nice, I've removed `expires`. Good call.

File: src/shared/utils/browser/set-auth-cookie.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { isHttps } from "@utils/env";
+import * as cookie from "cookie";
+import { authCookieName } from "../../config";
+
+export default function setAuthCookie(jwt: string) {
+  const expires = new Date();
+  expires.setDate(expires.getDate() + 365);
+
+  document.cookie = cookie.serialize(authCookieName, jwt, {
+    expires,
```
Comment (SleeplessOne1917) [inline src/shared/utils/browser/set-auth-cookie.ts:10]: This still uses `expires`. Judging by the article I linked earlier, `expires` has more backward compatibility than `maxAge` (although it seems only some versions of internet explorer don't support `maxAge` - not sure if it's worth bothering). Either way, if `clearAuthCokkie` is using `maxAge`, this should as well. `maxAge` is in milliseconds.

**Review by SleeplessOne1917 — CHANGES_REQUESTED** [review]

File: src/shared/utils/browser/set-auth-cookie.ts (hunk 4 above)
Comment (alectrocute) [reply src/shared/utils/browser/set-auth-cookie.ts:10]: Good point. Is there any way you could quickly push a commit to fix that here? I'm about to be out of office for the rest of the day.

**Review by SleeplessOne1917 — APPROVED** [review]

**Pull request merged**